### ISSUE TITLE
Trying to get all sub pages of a page through a custom tag does not work

### ISSUE THREAD

**Issue by kolaente** [statamic/cms#3774]
## Bug Description

I've created a custom tag to return me an entry collection of all sub pages of a page (see below). The goal is to show a nice list of all sub pages of a particular page. These are pages of a collection and I don't want to show all of them at once, hence I'm not using the `{{ collection }}` tag already shipped with statamic.

Querying all entries by their parent as demonstrated in the code below fails with an exception in `\Statamic\Query\Builder::filterTestEquals`: "strtolower() expects parameter 1 to be string, object given"
I can see in the debugger that the `$item` passed to that function is a `\Statamic\Structures\Page` and the `$value` is the page param. I guess the error kind of makes sense, I just don't understand why it happened. This seems to happen only with the `parent` filter in the where condition. Any other condition seems to work fine, even nonexisting ones.

## How to Reproduce

Create a new tag like this:

```php
<?php

namespace App\Tags;

use Statamic\Facades\Entry;
use Statamic\Tags\Tags;

class ChildPages extends Tags
{
    /**
     * The {{ child_pages }} tag.
     * It will return a collection of all pages which are the parent of the currently visited page.
     *
     * @return string|array
     */
    public function index()
    {
        $parent = $this->params->get('page');
       return  Entry::query()
            ->where('parent', $parent)
            ->get();
    }
}
```

Then call it something like this in an antlers template: 
```
    {{ child_pages page="95ef0a90-9ec5-4063-b356-3ef4a3c2c462" }}
        {{ partial:entry }}
    {{ /child_pages }}
```

## Environment

Statamic 3.1.16 Pro
Laravel 8.43.0
PHP 7.4.16
kolaente/statamic-snippet 1.1.1

**Install method (choose one):** Fresh install from `statamic/statamic`

**Comment by kolaente** [comment]
I ended up just getting all entries from the collection and filtering everything I did not need with a laravel collection filter closure:

```php
return Collection::find('pages')
    ->queryEntries()
    ->get()
    ->filter(function ($value, $args) {
        return $value->parent() && $value->parent()->id() === $this->params->get('page');
    });
```

From a database perspective, this is probably not the most ideal solution but that collection only has 30ish entries and the result gets cached anyway so it should be fine.

**Comment by webmalocher** [comment]
Using `{{ collection from="the_collection" :parent:is="id" }}` results in the same behavior as described above, even if I use a static value for the parent ID.

**Comment by brenjt** [comment]
Why can't you just do this:

```
{{ collection from="the_collection" }}
{{ if parent:id === page:id }}
    ...
{{ /if }}
{{ /collection }}
```

**Comment by kolaente** [comment]
@brenjt I guess that would've worked. Huh. 

Never thought of doing it that way tbh, I guess that comes from looking for a "clean" solution instead of the fastest one to get results.

**Comment by jasonvarga** [comment]
I believe this has been fixed in 3.2.22. You should be able to do `{{ collection:pages :parent:is="id" }}`

**Comment by kolaente** [comment]
@jasonvarga That seems to work well for collections, the solution I ended up with also works for non-collections so I'll keep using it. Thanks for the hint!
